src/js/ui/search/searchView.tsx

```diff
@@ -0,0 +1,238 @@
+import { renderPlayer, renderBoard, getButton } from '../user/games/gamesView';
+import * as helper from '../helper';
+import * as gameApi from '../../lichess/game';
+import gameStatus from '../../lichess/status';
+import session from '../../session';
+import * as utils from '../../utils';
+import router from '../../router';
+
+export default function view(vnode: Mithril.Vnode<{}, SearchState>) {
+  const ctrl = vnode.state;
+
+  function header() {
+    return headerWidget(i18n('search'));
+  }
+
+  return layout.free(header, () => renderSearchForm(ctrl));
+}
+
+function renderSearchForm(ctrl: SearchState) {
+  const ratingOptions = settings.search.ratings.map((a: string) => ({value: a, label: a}));
+  const opponents = [['0', i18n('human') + ' ' + i18n('opponent')], ['1', i18n('computer') + ' ' + i18n('opponent')]];
+  const opponentOptions = opponents.map((a: Array<string>) => ({value: a[0], label: a[1]}));
+  const aiLevelOptions = settings.search.aiLevels.map((a: string) => ({value: a, label: i18n('level') + ' ' + a}));
+  const sourceOptions = settings.search.sources.map((a: Array<string>) => ({value: a[0], label: a[1]}));
+  const perfOptions = settings.search.perfs.map((a: Array<string>) => ({value: a[0], label: a[1]}));
+  const turnOptions = settings.search.turns.map((a: string) => ({value: a, label: a}));
+  const durationOptions = settings.search.durations.map((a: Array<string>) => ({value: a[0], label: a[1]}));
+  const timeOptions = settings.search.times.map((a: Array<string>) => ({value: a[0], label: a[1]}));
+  const incrementOptions = settings.search.increments.map((a: Array<string>) => ({value: a[0], label: a[1]}));
+  const resultOptions = settings.search.results.map((a: Array<string>) => ({value: a[0], label: a[1]}));
+  const winnerOptions = settings.search.winners.map((a: Array<string>) => ({value: a[0], label: i18n(a[1])}));
+  const dateOptions = settings.search.dates.map((a: Array<string>) => ({value: a[0], label: a[1]}));
+  const sortFieldOptions = settings.search.sortFields.map((a: Array<string>) => ({value: a[0], label: a[1]}));
+  const sortOrderOptions = settings.search.sortOrders.map((a: Array<string>) => ({value: a[0], label: a[1]}));
+  const viewport = helper.viewportDim();
+  const boardsize = (viewport.vw - 20) * 0.30;
+  const boardBounds = { height: boardsize, width: boardsize };
+
+  return (
+    <div className="native_scroller searchWraper">
+      <form id="advancedSearchForm"
+      onsubmit={function(e: Event) {
+        e.preventDefault();
+        const analysed = document.getElementById('analysed') as HTMLInputElement;
+        analysed.value = analysed.checked ? '1' : '';
+        return ctrl.search(e.target as HTMLFormElement);
+      }}>
+          <div className="game_search_row">
+            <label>Players: </label>
+            <div className="game_search_input">
+              <input type="text" id="players_a" name="players.a" onkeyup={(e: Event) => { redraw(); }} />
```

Comment:
prefer `oninput` to `onkeyup`. No use to declare function arg if you don't use it: `() => redraw()` is simpler

it is also useful to throttle this input handler